src/render_engine/dx12/dx12_render_engine.cpp

```diff
@@ -306,34 +306,43 @@ namespace nova::renderer::rhi {
         // We want to create descriptors in the heaps in the order of their bindings
         for(const DescriptorSetWrite& write : writes) {
             const auto* set = static_cast<const DX12DescriptorSet*>(write.set);
+
             CD3DX12_CPU_DESCRIPTOR_HANDLE write_handle = CD3DX12_CPU_DESCRIPTOR_HANDLE(set->heap->GetCPUDescriptorHandleForHeapStart(),
-                                                                                       cbv_srv_uav_descriptor_size * write.binding);
+                                                                                       cbv_srv_uav_descriptor_size * write.first_binding);
 
             switch(write.type) {
                 case DescriptorType::CombinedImageSampler: {
-                    const auto* image = static_cast<const DX12Image*>(write.image_info.image);
-
-                    D3D12_SHADER_RESOURCE_VIEW_DESC srv_desc = {};
-                    srv_desc.Format = to_dxgi_format(write.image_info.format.pixel_format);
-                    srv_desc.ViewDimension = D3D12_SRV_DIMENSION_TEXTURE2D;
-                    srv_desc.Texture2D.MostDetailedMip = 0;
-                    srv_desc.Texture2D.MipLevels = 1;
-                    srv_desc.Texture2D.PlaneSlice = 0;
-                    srv_desc.Texture2D.ResourceMinLODClamp = 0;
-
-                    device->CreateShaderResourceView(image->resource.Get(), &srv_desc, write_handle);
-
+                    for(uint32_t i = 0; i < write.bindings.size(); i++) {
+                        const auto& binding = write.bindings.at(i);
```

Comment:
Should this be `[]`? It's up to you, but this seems a pretty cut and dry valid access.

Reply:
I like `.at` because it's const and it'll never add to the vector, but `[]` might add to the vector.